### PR TITLE
gh-107526: Revert "gh-100357: Convert several functions in bltinsmodule to AC"

### PR DESCRIPTION
**Pull request by sobolevn**
* Issue: gh-107526

### CODE REVIEW

File: Python/bltinmodule.c [hunk 1]
```diff
@@ -2444,36 +2437,33 @@ builtin_sorted(PyObject *self, PyObject *const *args, Py_ssize_t nargs, PyObject
...
 {
+    PyObject *v = NULL;
     PyObject *d;
 
-    if (object == NULL) {
-        d = _PyEval_GetFrameLocals();
+    if (!PyArg_UnpackTuple(args, "vars", 0, 1, &v))
+        return NULL;
+    if (v == NULL) {
+        d = Py_XNewRef(PyEval_GetLocals());
```
Comment (JelleZijlstra) [inline Python/bltinmodule.c]: This should use `_PyEval_GetFrameLocals` (I think that's what's causing the test failures; maybe there's similar issues in `dir()` and `locals()`).

Comment (sobolevn) [reply Python/bltinmodule.c]: Thanks, I've missed it.

**Review by JelleZijlstra — APPROVED** [review]

**Comment by JelleZijlstra** [comment]
@sobolevn could you make a 3.12 backport?

**Comment by xncbf** [comment]
It seems that the workflow is broken starting from this PR.
https://github.com/python/cpython/actions/workflows/build.yml

**Comment by hugovk** [comment]
Triage: Is the 3.12 backport still needed? If not, let's remove the label.

**Pull request merged**